### ISSUE TITLE
Make Optics into scoped extension functions

### ISSUE THREAD

**Issue by pakoito** [arrow-kt/arrow#866]
For example, reencode https://github.com/arrow-kt/arrow/blob/master/modules/optics/arrow-optics/src/main/kotlin/arrow/optics/Optional.kt#L239 as

```
fun S.getOrModify(): Either<T, A>

fun S.modify(f: (A) -> B): T = getOrModify().fold(::identity, { a -> set(s, f(a)) })
```

That way you can write

```
Employees.employees.every(ListK.each()).company.address.street.name.run {
  employees.modify(String::capitalize)
}
```

**Comment by raulraja** [comment]
I think we should provide alternative functions, meaning exposing it both ways since the snippet below is valid in Arrow and considered good style as well. There are sometimes you may not want the extra nesting and indirection just to achieve `.` property syntax.
```kotlin
ListK.each<Employee>().run {
  Employees.employees.every.company.address.street.name.modify(employees, String::capitalize)
}
```
Another alternative would be:
```kotlin
ListK.each<Employee>().run {
  Employees.employees.every.company.address.street.name.modify(String::capitalize)(employees)
}
```

**Comment by nomisRev** [comment]
@pakoito I agree with @raulraja that I'd prefer to enable both by adding alternative functions. As I've expressed earlier I sometimes miss the old API of type classes such as `FF.map(fa, f)` but that could be a force of habit.

Side-note that the second snippet uses a method called `lift`. Original reasoning was that you can use it to lift a function `(String) -> String` to `(Person) -> Person` using a `Lens<Person, String>`. Do we still feel `lift` is the correct naming? AFAIK Scala (Cats) / Haskell prefers curried signatures but we don't. Do we feel comfortable with this naming still?

**Comment by nomisRev** [comment]
@pakoito @raulraja let's re-open if we want to discuss this again.
